src/nvim/message.c

```diff
@@ -3022,6 +3022,13 @@ void msg_clr_eos_force(void)
     msg_row = msg_grid_pos;
   }
 
+  if (need_clr_eos && msg_col > 0 && msg_grid.chars[0] != '-') {
```

Comment:
Won't this break for multiline messages, messages that start with `-`, and cmdline drawn from right to left?

Reply:
maybe. i thought a new way .i will rewrite here after i wake up
 🙃